src/Formik.tsx

```diff
@@ -137,24 +138,32 @@ export function useFormik<Values = object>({
 
   React.useEffect(
     () => {
-      if (!!validateOnChange && !state.isSubmitting) {
+      if (!!validateOnChange && !state.isSubmitting && isMounted.current) {
         return runValidationAsEffect();
       }
       return;
     },
-    [state.values, state.isSubmitting, validateOnChange]
+    [state.values]
   );
 
   React.useEffect(
     () => {
-      if (!!validateOnBlur && !state.isSubmitting) {
+      if (!!validateOnBlur && !state.isSubmitting && isMounted.current) {
```

Comment:
Checking `isMounted` in `useEffect` really made me cringe but I couldn't find an easy way around it currently.